errbot/utils.py

```diff
@@ -146,7 +147,7 @@ def find_roots(path, file_sig='*.plug'):
             the file pattern to look for
        :return: a set of paths
     """
-    roots = set()  # you can have several .plug per directory.
```

Comment:
No, you have ignored the comment, you need to setup them right?

Reply:
I am not sure I understand your question, but let me try to get it right.

Before, a `set` was used to return a unique list of paths in which to look for plugins. The problem with using a `set` is that it is unordered. I replaced the set with a `list`. This means that the paths will be added multiple times (as stated by the comment).

In order to make the list unique again and at the same same preserve order, I use an `OrderedDict`. I moved the original comment to the `return` statement to indicate this:

```python
    return list(collections.OrderedDict.fromkeys(roots))  # you can have several .plug per directory
```

If the purpose of the original `set` was to ensure the list was unique, then this PR should not change that. It just assures the order is always the same.

Am I missing something?

Reply:
Doh sorry about, thanks for the patch!